### PR TITLE
Add GCS uri support for loading and saving checkpoints

### PR DESCRIPTION
**Pull request by eracah**
# What does this PR do?

Adds ability to load and save checkpoints from GCS using a `gs://` uri

Added unit tests
Ran manual tests
Added docs

# What issue(s) does this change relate to?

CO-1490

# Before submitting
- [x] Was this change discussed/approved in a GitHub issue first? It is much more likely to be merged if so.
- [x] Did you update any related docs and document your change?
- [x] Did you update any related tests and add any new tests related to your change? (see [testing](https://github.com/mosaicml/composer/blob/dev/CONTRIBUTING.md#running-tests))
- [x] Did you run the tests locally to make sure they pass?
- [x] Did you run `pre-commit` on your change? (see the `pre-commit` section of [prerequisites](https://github.com/mosaicml/composer/blob/dev/CONTRIBUTING.md#prerequisites))

### CODE REVIEW

File: composer/utils/file_helpers.py [hunk 1]
```diff
@@ -345,6 +345,17 @@ def maybe_create_object_store_from_uri(uri: str) -> Optional[ObjectStore]:
     elif backend == 'wandb':
         raise NotImplementedError(f'There is no implementation for WandB load_object_store via URI. Please use '
                                   'WandBLogger')
+    elif backend == 'gs':
+        if 'GCS_KEY' not in os.environ or 'GCS_SECRET' not in os.environ:
```
Comment (dakinggg) [inline composer/utils/file_helpers.py:349]: Are there standard environment variables we should be using for this? Like what does MCLOUD do for GCS auth? (What you have is probably correct, just checking)

**Review by dakinggg — APPROVED** [review]
LGTM, comments minor stuff like docs and env var naming

File: composer/utils/file_helpers.py (hunk 1 above)
Comment (eracah) [reply composer/utils/file_helpers.py:349]: Yeah exactly streaming and mcli use those env variables

File: docs/source/trainer/checkpointing.rst [hunk 2]
```diff
@@ -299,9 +299,9 @@ and then the :class:`.RemoteUploaderDownloader` logger will upload checkpoints t
 
 Behind the scenes, the :class:`.RemoteUploaderDownloader` uses :doc:`Apache Libcloud <libcloud:storage/index>`.
 
-The easiest way to upload checkpoints to S3 or OCI is to prefix your ``save_folder``  with ``'s3://'`` or ``'oci://'``. All other
+The easiest way to upload checkpoints to S3, OCI, or GCS is to prefix your ``save_folder``  with ``'s3://'``, ``'oci://'``, or ``'gs://'``. All other
```
[the remark this answers is not in the document]
Comment (eracah) [reply docs/source/trainer/checkpointing.rst]: fixed

File: docs/source/trainer/checkpointing.rst [hunk 3]
```diff
@@ -351,67 +351,43 @@ and delete the checkpoints from the local disk. The checkpoints will be located
...
 
     trainer.fit()
 
-.. seealso::
-
-    *   :doc:`Full list of object store providers <libcloud:storage/supported_providers>`
-    *   :class:`~.RemoteUploaderDownloader`
-
-
+This will train your model, saving the checkpoints locally, upload them to the GCS Bucket `my_bucket`,
```
[the remark this answers is not in the document]
Comment (eracah) [reply docs/source/trainer/checkpointing.rst]: fixed

File: docs/source/trainer/checkpointing.rst [hunk 4]
```diff
@@ -523,8 +462,8 @@ should be the path to the checkpoint file *within the container/bucket*.
 
     new_trainer.fit()
 
-An easier way to load checkpoints from S3 or OCI specifically is to just use a URI starting with ``s3://`` or ``oci://``.
-If you use the S3 or OCI URI, it is not necessary to specify a ``load_object_store``. Note, that for other
+An easier way to load checkpoints from S3, OCI, GCS specifically is to just use a URI starting with ``s3://``, ``oci://``, or ``gs://``.
+If you use the S3, OCI, or GCS URI, it is not necessary to specify a ``load_object_store``. Note, that for other
 object stores like WandB or LibCloud, you must still specify a ``load_object_store``.
 
 .. testcode::
```
[the remark this answers is not in the document]
Comment (eracah) [reply docs/source/trainer/checkpointing.rst:475]: fixed. they just needed a new line

File: docs/source/trainer/checkpointing.rst [hunk 5]
```diff
@@ -299,9 +299,9 @@ and then the :class:`.RemoteUploaderDownloader` logger will upload checkpoints t
...
 Behind the scenes, the :class:`.RemoteUploaderDownloader` uses :doc:`Apache Libcloud <libcloud:storage/index>`.
 
-The easiest way to upload checkpoints to S3 or OCI is to prefix your ``save_folder``  with ``'s3://'`` or ``'oci://'``. All other
+The easiest way to upload checkpoints to S3, OCI, or GCS is to prefix your ``save_folder``  with ``'s3://'``, ``'oci://'``, or ``'gs://'``. All other
 checkpoint arguments remain the same. For example, ``save_filename`` will be the name of the checkpoint file
-that gets uploaded to the S3 or OCI URI that you specified.
+that gets uploaded to the S3, OCI, or GCS URI that you specified.
 
 For example, for S3:
 .. testcode::
```
[the remark this answers is not in the document]
Comment (eracah) [reply docs/source/trainer/checkpointing.rst:308]: fixed. it just needs a new line between the "for example..." and the test code

**Pull request merged**